Resources/Prototypes/Entities/Structures/Power/apc.yml

```diff
@@ -93,6 +93,62 @@
         acts: [ "Destruction" ]
   - type: StationInfiniteBatteryTarget
 
+# APC under construction
+- type: entity
+  noSpawn: true
+  id: APCFrame
+  name: APC Frame
+  suffix: No electronics
+  description: A control terminal for the area's electrical systems, lacking the electronics.
+  placement:
+    mode: SnapgridCenter
+  components:
+  - type: Clickable
+  - type: InteractionOutline
+  - type: Transform
+    anchored: true
+  - type: Sprite
+    drawdepth: WallMountedItems
+    netsync: false
+    sprite: Structures/Power/apc.rsi
+    state: apcframe
+  - type: Construction
+    graph: APC
+    node: apcFrame
+  - type: WallMount
+  - type: Damageable
+    damageContainer: Inorganic
+    damageModifierSet: Metallic
+  - type: Destructible
+    thresholds:
+    - trigger:
+        !type:DamageTrigger
+        damage: 200
+      behaviors: #excess damage, don't spawn entities.
+      - !type:DoActsBehavior
+        acts: [ "Destruction" ]
+    - trigger:
+        !type:DamageTrigger
+        damage: 50
+      behaviors:
+      - !type:SpawnEntitiesBehavior
+        spawn:
+          SheetSteel1:
+            min: 1
+            max: 1
+      - !type:DoActsBehavior
+        acts: [ "Destruction" ]
+
+# Constructed APC
+- type: entity
+  parent: BaseAPC
+  id: APCConstructed
+  suffix: Open
+  components:
+  - type: Apc
+    voltage: Apc
+    open: true
```

Comment:
This can also be done by adding an IGraphAction to the construction graph that opens the panel, but this should be fine.

Reply:
I tried doing it that way, but ran into the problem that either I had the IGraphAction called on the construction step like this:

>  -  node: apcFrame
>     edges:
>      -  to: apc
>         completed:
>            -  !type:OpenPanel
>               open: true

Which then simply didnt seem to be called on the new apc entity, only the old entity

or the IGraphAction was called on the node like this:

> - node: apc
>    actions:
>     - !type:OpenPanel
>                open: true

Which had the problem that all apc:s started out with their panel opened, which made me ask in discord for how I could perhaps check if the object was manually constructed/placed on the map to begin with/spawned in/etc, and I got the tip to just have another entity with the panel open (like I did now), which is what worked for me

Reply:
Ah yeah, I ran into that while messing with construction graphs a few days ago too. Looks good then. Great job!